File: react_native/src/domain/repositories/UrlRepository.ts
import { CleanedUrl } from '../entities/CleanedUrl';

export interface UrlStatistics {
  totalUrls: number;
  totalCharactersSaved: number;
  totalTrackersRemoved: number;
}

export interface UrlRepository {
  parseUrl(url: string): Promise<CleanedUrl>;
  saveToHistory(cleanedUrl: CleanedUrl): Promise<void>;
  getHistory(): Promise<CleanedUrl[]>;
  clearHistory(): Promise<void>;
  getStatistics(): Promise<UrlStatistics>;
}

File: flutter/lib/domain/repositories/url_repository.dart
import '../entities/cleaned_url.dart';

abstract class UrlRepository {
  Future<CleanedUrl> parseUrl(String url);
  Future<void> saveToHistory(CleanedUrl cleanedUrl);
  Future<List<CleanedUrl>> getHistory();
  Future<void> clearHistory();
  Future<Map<String, int>> getStatistics();
}